internal/kubernetes/event/event.go

```diff
@@ -140,3 +145,92 @@ func (e *sink) Patch(event *corev1.Event, data []byte) (*corev1.Event, error) {
 	err := e.client.Patch(e.ctx, event, libClient.RawPatch(types.StrategicMergePatchType, data))
 	return event, err
 }
+
+// NewPromotionEventAnnotations returns annotations for a Promotion related event.
+// It may skip some fields when error occurred during serialization, to record event with best-effort.
+func NewPromotionEventAnnotations(
```

Comment:
Please move this to `internal/event/promotion.go`.

Reply:
Plus maybe rename it to `NewPromotionAnnotations`, as `event` already should be enough to indicate it's about events.

Reply:
Additional candidate to be moved from the public API package is `FormatEventUserActor` (-> `FormatUserActor`).

Reply:
@hiddeco Thank you for the suggestion! I've moved this function to `internal/event/promotion.go`.

> Additional candidate to be moved from the public API package is FormatEventUserActor (-> FormatUserActor).

I've tried this but it still caused the import cycle issue like:
```
0.602   imports github.com/akuity/kargo/api/v1alpha1
0.602   imports github.com/akuity/kargo/internal/event
0.602   imports github.com/akuity/kargo/api/v1alpha1: import cycle not allowed
```
I think we could only move `NewPromotionAnnotations` here for minimal changes. We can discuss further improvements if necessary and I'd be very happy to contribute more.

Reply:
That's fair.